cmd/snap-bootstrap/partition/sfdisk.go

```diff
@@ -149,7 +149,12 @@ func ensureNodesExistImpl(ds []DeviceStructure, timeout time.Duration) error {
 			}
 			time.Sleep(100 * time.Millisecond)
 		}
-		if !found {
+		if found {
+			output, err := exec.Command("udevadm", "trigger", "--settle", part.Node).CombinedOutput()
```

Comment:
`udevadm trigger --settle` works for systemd 238 and later+, 18.04 comes with 237. You'll have to break this into separate `udevadm trigger`, (maybe `--subsystem-match=block` too?), followed by `udevadm settle --timeout=<sometimeout>`, IIRC we're using `--timeout=10` in snapd.

Reply:
This code is targetting focal & UC20 only right? thus systemd v238+ is available....

Reply:
In that case we're fine.

Reply:
Err, almost, the spread test that failed needs to be changed to run on 19.10.

Reply:
I did change the snap-recovery test to run on 19.10 but forgot to do the same to snap-recovery-autodetect. Fixed now.

Reply:
FWIW, does it make sense to try `--subsystem-match=block` ?

Reply:
Forget it, missed that we're passing the device.

Reply:
We're already specifying a device target so also specifying a subsystem would AFAICT trigger udev for all devices matching the subsystem, in each invocation.